pkg/config/config.go

```diff
@@ -25,21 +38,75 @@ func Read(filename string) (*Config, error) {
 
 // NewYAML parses the given YAML data into a Config.
 func NewYAML(input []byte) (*Config, error) {
+	var inputYAML configpb.Config
 	// Parse the raw YAML into a structure.
-	var messages custom_detectorspb.CustomDetectors
-	if err := protoyaml.UnmarshalStrict(input, &messages); err != nil {
+	if err := protoyaml.UnmarshalStrict(input, &inputYAML); err != nil {
 		return nil, err
 	}
-	// Convert the structured YAML into detectors.
-	var d []detectors.Detector
-	for _, detectorConfig := range messages.Detectors {
+
+	// Convert to detectors.
+	var detectorConfigs []detectors.Detector
+	for _, detectorConfig := range inputYAML.Detectors {
 		detector, err := custom_detectors.NewWebhookCustomRegex(detectorConfig)
 		if err != nil {
 			return nil, err
 		}
-		d = append(d, detector)
+		detectorConfigs = append(detectorConfigs, detector)
+	}
+
+	// Convert to configured sources.
+	var sourceConfigs []sources.ConfigurableSource
+	for _, pbSource := range inputYAML.Sources {
+		s, err := instantiateSourceFromType(pbSource.GetType())
+		if err != nil {
+			return nil, err
+		}
+		src := sources.NewConfigurableSource(s)
+		src.Configure(pbSource)
+
+		sourceConfigs = append(sourceConfigs, src)
 	}
```

Comment:
Pre-configuring the sources like this has a few implications:
1. We lose access to the protobuf configuration (the data is hidden via the `ConfigurableSource`)
2. Validation of the config file happens close to parsing it
3. Because of 2, you can't put unused garbage in the config file (e.g. if you just want to use the config file for custom detectors)